roles/setup-services/tasks/main.yml

```diff
@@ -0,0 +1,85 @@
+---
+
+- name: Check namespace doesn't already exist
+  shell: oc get namespace {{ namespace }}
+  register: namespace_exists
+  failed_when: namespace_exists.stderr != '' and 'NotFound' not in namespace_exists.stderr
+
+- name: Create namespace {{ namespace }} if it doesnt exist
+  shell: oc create namespace {{ namespace }}
+  when: namespace_exists.rc != 0
+
+- include_tasks: provision-service.yml
+  with_items: "{{ services }}"
+
+- name: Wait for all services to be ready
+  shell: oc get serviceinstance -o=jsonpath='{.items[*].status.conditions[0].status}' | tr ' ' '\n' | grep True | wc -w
+  register: ready_status_output
+  until: ready_status_output.stdout == "6"
+  retries: 50
+  delay: 10
```

Comment:
I would recommend to increase the number of retries or delay. When I gave it a try on integreatly cluster, it took 49 retries before it completed, but I had the apb images pre-pulled.